woocommerce/woocommerce-gutenberg-products-block
# Revert "Allow LegacyTemplate block to be reinserted, only on WooCommerce block templates."

Reverts woocommerce/woocommerce-gutenberg-products-block#5545 as this PR introduced this issue https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5636

You can read more about why we've reverted this here https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5637#issuecomment-1022332802

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5636

### Manual Testing

This PR includes regression testing around the feature due to the nature of the bug and its fix.

1. Checkout this branch, run `npm run build` and start outside of the Site Editor e.g. Appearance > Themes
2. Load Appearance > Site Editor and select one of the WooCommerce templates. Confirm it loads the correct Legacy Template block in the Editor for the given block template.
3. Remove this block, you **should not** be able to reinsert it and **do not save** (this is expected as part of this revert)
4. Refresh the page and confirm the correct Legacy Template block still gets rendered in the Site Editor
5. Customise this template, and save it. Now reload this template and ensure it loads the customised template in the Site Editor and on the frontend.
6. Try clearing the customisations on Site Editor > Templates screen, ensure these are reset in the Site Editor and on the frontend.

Try the above steps with another WooCommerce template.

## Changelog
> Fix a bug with allowing the LegacyTemplate block to be reinserted, only on WooCommerce block templates
